nodes/ProgramNode.java

```diff
@@ -43,6 +45,22 @@ public String convertToPython() {
 
     @Override
     public boolean validateTree() {
+        // TODO this could be moved elsewhere if needed
```

Comment:
good catch, I think it's good to leave here. Makes most sense to be at the top level imo